packages/typescript-vfs/src/index.ts

```diff
@@ -107,86 +107,90 @@ export const knownLibFilesForCompilerOptions = (compilerOptions: CompilerOptions
   const lib = compilerOptions.lib || []
 
   // Note that this will include files which can't be found for particular versions of TS
-  const files = [
-    "lib.d.ts",
-    "lib.decorators.d.ts",
-    "lib.decorators.legacy.d.ts",
-    "lib.dom.d.ts",
-    "lib.dom.iterable.d.ts",
-    "lib.es2015.collection.d.ts",
-    "lib.es2015.core.d.ts",
-    "lib.es2015.d.ts",
-    "lib.es2015.generator.d.ts",
-    "lib.es2015.iterable.d.ts",
-    "lib.es2015.promise.d.ts",
-    "lib.es2015.proxy.d.ts",
-    "lib.es2015.reflect.d.ts",
-    "lib.es2015.symbol.d.ts",
-    "lib.es2015.symbol.wellknown.d.ts",
-    "lib.es2016.array.include.d.ts",
-    "lib.es2016.d.ts",
-    "lib.es2016.full.d.ts",
-    "lib.es2017.d.ts",
-    "lib.es2017.full.d.ts",
-    "lib.es2017.intl.d.ts",
-    "lib.es2017.object.d.ts",
-    "lib.es2017.sharedmemory.d.ts",
-    "lib.es2017.string.d.ts",
-    "lib.es2017.typedarrays.d.ts",
-    "lib.es2018.asyncgenerator.d.ts",
-    "lib.es2018.asynciterable.d.ts",
-    "lib.es2018.d.ts",
-    "lib.es2018.full.d.ts",
-    "lib.es2018.intl.d.ts",
-    "lib.es2018.promise.d.ts",
-    "lib.es2018.regexp.d.ts",
-    "lib.es2019.array.d.ts",
-    "lib.es2019.d.ts",
-    "lib.es2019.full.d.ts",
-    "lib.es2019.intl.d.ts",
-    "lib.es2019.object.d.ts",
-    "lib.es2019.string.d.ts",
-    "lib.es2019.symbol.d.ts",
-    "lib.es2020.bigint.d.ts",
-    "lib.es2020.d.ts",
-    "lib.es2020.date.d.ts",
-    "lib.es2020.full.d.ts",
-    "lib.es2020.intl.d.ts",
-    "lib.es2020.number.d.ts",
-    "lib.es2020.promise.d.ts",
-    "lib.es2020.sharedmemory.d.ts",
-    "lib.es2020.string.d.ts",
-    "lib.es2020.symbol.wellknown.d.ts",
-    "lib.es2021.d.ts",
-    "lib.es2021.full.d.ts",
-    "lib.es2021.intl.d.ts",
-    "lib.es2021.promise.d.ts",
-    "lib.es2021.string.d.ts",
-    "lib.es2021.weakref.d.ts",
-    "lib.es2022.array.d.ts",
-    "lib.es2022.d.ts",
-    "lib.es2022.error.d.ts",
-    "lib.es2022.full.d.ts",
-    "lib.es2022.intl.d.ts",
-    "lib.es2022.object.d.ts",
-    "lib.es2022.regexp.d.ts",
-    "lib.es2022.sharedmemory.d.ts",
-    "lib.es2022.string.d.ts",
-    "lib.es2023.array.d.ts",
-    "lib.es2023.d.ts",
-    "lib.es2023.full.d.ts",
-    "lib.es5.d.ts",
-    "lib.es6.d.ts",
-    "lib.esnext.d.ts",
-    "lib.esnext.full.d.ts",
-    "lib.esnext.intl.d.ts",
-    "lib.esnext.promise.d.ts",
-    "lib.esnext.string.d.ts",
-    "lib.esnext.weakref.d.ts",
-    "lib.scripthost.d.ts",
-    "lib.webworker.d.ts",
-    "lib.webworker.importscripts.d.ts",
-  ]
+  const files =
+    "getAllLibFileNames" in ts
+      ? // @ts-ignore - see https://github.com/microsoft/TypeScript/pull/54011
+        (ts.getAllLibFileNames() as string[])
+      : [
```

Comment:
Does this list get filtered in any way to the actual set based on what's available? I'm not sure I totally know how this list is used.

Reply:
Yeah, there's a rough heuristic which cuts down the list based on `target`  + `lib` - just below:

```ts
  const targetToCut = ts.ScriptTarget[target]
  const matches = files.filter(f => f.startsWith(`lib.${targetToCut.toLowerCase()}`))
  const targetCutIndex = files.indexOf(matches.pop()!)

  const getMax = (array: number[]) =>
    array && array.length ? array.reduce((max, current) => (current > max ? current : max)) : undefined

  // Find the index for everything in
  const indexesForCutting = lib.map(lib => {
    const matches = files.filter(f => f.startsWith(`lib.${lib.toLowerCase()}`))
    if (matches.length === 0) return 0

    const cutIndex = files.indexOf(matches.pop()!)
    return cutIndex
  })

  const libCutIndex = getMax(indexesForCutting) || 0

  const finalCutIndex = Math.max(targetCutIndex, libCutIndex)
  return files.slice(0, finalCutIndex + 1)
}
```

So the order is important WRT downloading which have  some tests covering the cases, but the rough gist is that the CDN version will download all of the files (and cache them locally if possible) and put them into the VFS which TypeScript will then use synchronously later

I just added some checks for handle CDNs returning NOOPs for dts files which dont exist for that ts version